client/app/components/queries/SchemaBrowser.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { react2angular } from 'react2angular';
+import PropTypes from 'prop-types';
+import { Icon } from 'antd';
+import { debounce } from 'lodash';
+import { $rootScope } from '@/services/ng';
+
+import { Schema } from '../proptypes';
+
+const isSchemaEmpty = schema => schema === undefined || !schema.length;
```

Comment:
`isArray(schema) && (schema.length > 0)`:

1. it's self-explaining: we expect that schema is an array with non-zero length.
2. it's more reliable (what if schema will be null? object?)

Reply:
what about: 
`const isSchemaValid = schema => schema.constructor === Array && schema.length;`

Reply:
`isArray` === `schema.constructor === Array`, but it's shorter and literally means what we need to check - that argument "is array" (vs. "argument's constructor is equal to Array") - that's rather a question of semantics.